great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -274,7 +290,505 @@ def _determine_anomaly_coded_line(
+                tooltip=tooltip,
+            )
+            .properties(height=detail_line_chart_height)
+            .transform_filter(selection)
+        )
+
+        upper_limit: alt.Chart = (
+            alt.Chart(data=df)
+            .mark_line(color=line_color)
+            .encode(
+                x=alt.X(
+                    domain_name,
+                    type=domain_type,
+                    title=domain_title,
+                ),
+                y=alt.Y(max_value, type=metric_type, title=metric_title),
+                tooltip=tooltip,
+            )
+            .properties(height=detail_line_chart_height)
+            .transform_filter(selection)
+        )
+
+        band: alt.Chart = (
+            alt.Chart(data=df)
+            .mark_area()
+            .encode(
+                x=alt.X(
+                    domain_name,
+                    type=domain_type,
+                    title=domain_title,
+                ),
+                y=alt.Y(min_value, title=metric_title, type=metric_type),
+                y2=alt.Y2(max_value, title=metric_title),
+            )
+            .properties(height=detail_line_chart_height)
+            .transform_filter(selection)
+        )
+
+        # encode point color based on anomalies
+        predicate: alt.expr.core.BinaryExpression
+        if strict_min and strict_max:
+            predicate = (
+                (alt.datum.min_value > alt.datum[metric_name])
+                & (alt.datum.max_value > alt.datum[metric_name])
+            ) | (
+                (alt.datum.min_value < alt.datum[metric_name])
+                & (alt.datum.max_value < alt.datum[metric_name])
+            )
+        elif strict_min:
+            predicate = (
+                (alt.datum.min_value > alt.datum[metric_name])
+                & (alt.datum.max_value >= alt.datum[metric_name])
+            ) | (
+                (alt.datum.min_value < alt.datum[metric_name])
+                & (alt.datum.max_value <= alt.datum[metric_name])
+            )
+        elif strict_max:
+            predicate = (
+                (alt.datum.min_value >= alt.datum[metric_name])
+                & (alt.datum.max_value > alt.datum[metric_name])
+            ) | (
+                (alt.datum.min_value <= alt.datum[metric_name])
+                & (alt.datum.max_value < alt.datum[metric_name])
+            )
+        else:
+            predicate: alt.expr.core.BinaryExpression = (
+                (alt.datum.min_value >= alt.datum[metric_name])
+                & (alt.datum.max_value >= alt.datum[metric_name])
+            ) | (
+                (alt.datum.min_value <= alt.datum[metric_name])
+                & (alt.datum.max_value <= alt.datum[metric_name])
+            )
+        point_color_condition: alt.condition = alt.condition(
+            predicate=predicate,
+            if_false=alt.value(Colors.GREEN.value),
+            if_true=alt.value(Colors.PINK.value),
+        )
```

Comment:
Could some of these longer functions be refactored into helpers? I know Altair is quite verbose by default but I think there's room for some increment refactoring (perhaps after this PR).